gdnative-derive/src/lib.rs

```diff
@@ -237,18 +214,19 @@ pub fn profiled(meta: TokenStream, input: TokenStream) -> TokenStream {
 ///
 /// **Important**: This needs to be added to one and only one `impl` block for a given `NativeClass`.
 ///
-/// For additional details about how `#[methods]` expands, please refer to [gdnative::methods](macro@methods)
-///
 /// ### `#[method]`
 /// Registers the attributed function signature to be used by Godot.
 ///
 /// This attribute was formerly called `#[export]`, but is not directly related to the concept of
 /// [exporting](https://docs.godotengine.org/en/stable/tutorials/export/exporting_basics.html) in GDScript.
 ///
 /// A valid function signature must have:
-/// - `&self` or `&mut self` as its first parameter
-/// - Optionally, `&T` or `TRef<T>` where T refers to the type declared in `#[inherit(T)]` attribute as it's second parameter;
-///   this is typically called the _base_. The parameter must be attributed with `#[base]`.
+/// - `self`, `&self` or `&mut self` as its first parameter, if applicable.
+/// - Up of one of each of the following special arguments, in any order, denoted by the attributes:
+///     - `#[base]` - A reference to the base/owner object. This may be `&T` or `TRef<T>`m where `T` refers to
+///       the type declared in `#[inherit(T)]` attribute for the `NativeClass` type.
+///     - `#[ctx]` - The [async context](gdnative::tasks::Context), for async methods. See the `async` argument
+///       below.
```

Comment:
Typical parameter declaration is
```rs
#[ctx] ctx: Arc<Context>
```

Is that clear enough? Or should the type perhaps be named `AsyncContext`? The latter would make sure that it doesn't make sense in a non-async function signature.

Reply:
Ah true, this type was already there. `#[async_ctx]` was actually my first idea, but I thought in the type it might be clearer.

`#[async_ctx]` _might_ be a bit clearer, but maybe we should ask some users. On the other hand, long signatures might be broken into multiple lines by rustfmt, and then having the same length as `#[opt]` would be a tiny readability bonus 😉probably not that much difference in practice.

Reply:
Changed to `#[async_ctx]` for now.